src/coreclr/jit/emitxarch.cpp

```diff
@@ -205,9 +205,15 @@ bool emitter::IsEvexEncodedInstruction(instruction ins) const
 // Returns:
 //    `true` if ins is a SIMD instruction.
 //
-bool emitter::IsVexOrEvexEncodedInstruction(instruction ins) const
+bool emitter::IsVexOrEvexEncodableInstruction(instruction ins) const
 {
-    return IsVexEncodedInstruction(ins) || IsEvexEncodedInstruction(ins);
+    if (!UseVEXEncoding())
+    {
+        return false;
+    }
+
+    insFlags flags = CodeGenInterface::instInfo[ins];
+    return (flags & (Encoding_VEX | Encoding_EVEX)) != 0;
 }
```

Comment:
This is a little bit faster than checking them independently. We don't need to check `UseEVEXEncoding()` since we'll end up filtering by `TakesEvexPrefix(id)` which in turn duplicates the `IsEvexEncodableInstruction` check.

We can't check just `VEX` since we need to also return `true` for `EVEX` only instructions.